judo-runtime-core-dao-rdbms/src/main/java/hu/blackbelt/judo/runtime/core/dao/rdbms/executors/DeleteStatementExecutor.java

```diff
@@ -101,17 +94,21 @@ public void executeDeleteStatements(NamedParameterJdbcTemplate jdbcTemplate,
         // Collect all information required to build dependencies between nodes.
         Set<RdbmsReference<ID>> deleteRdbmsReferences = toRdbmsReferences(deleteStatements, removeReferenceStatements);
 
+        AsmUtils asmUtils = new AsmUtils(getAsmModel().getResourceSet());
         toDependencySortedDeleteStatementStream(deleteStatements, deleteRdbmsReferences)
                 .forEach(consumer(deleteStatement -> {
 
                     EClass entity = deleteStatement.getInstance().getType();
                     ID identifier = deleteStatement.getInstance().getIdentifier();
 
-                    // Collecting all tables on the inheritance chain which required to insert.
-                    Stream.concat(
-                            ImmutableList.of(entity).stream(),
-                            entity.getEAllSuperTypes().stream()).filter(t -> isEntityType(t))
-                            .forEach(entityForCurrentStatement -> {
+                    // Collecting all tables on the inheritance chain to delete.
+                    Set<EClass> types = asmUtils.all(EClass.class).filter(c -> c.getEAllSuperTypes().contains(entity)).collect(Collectors.toSet());
+                    types.addAll(entity.getEAllSuperTypes());
+                    types.add(entity);
```

Comment:
Maybe better if AsmUtil have method where descandences can be query for an EClass. This recursive superclass collection for every class can be time consuming on every delete operation.

Reply:
And isEntityType check is missing.